### PR TITLE
Add guide to contribution

### LINKED ISSUE #330: Write Guide to Contribution

### PR DESCRIPTION
**Pull request by lorenaqendro**
Fixes #330

### Description
Added guide to contribution.

### Checklist

- [ ] ~~Created new test(s) (if applicable)~~
- [x] Updated the README / documentation (if applicable)

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+start with the [system architecture](documentation/System-Architecture.md) and [local datastore](documentation/Local-Datastore.md), and 
+eventually go further in the documentation with other topics of your interest.
+
+## Help wanted
+[Open Source Community - Digital Connector](https://github.com/FutureCitiesCatapult/TomboloDigitalConnector/milestone/22)
+milestone contains issues we think are interesting for you to work on. You can find them also through 
+[`help wanted`](https://github.com/FutureCitiesCatapult/TomboloDigitalConnector/issues?q=is%3Aopen+is%3Aissue+label%3A%22help+wanted%22) 
+tag. Feel free to check the other milestones in case you find something that excites you more.
+If none of the issues reflects your problem/feature create a new one following the 
+[issue template](https://github.com/FutureCitiesCatapult/TomboloDigitalConnector/blob/master/.github/ISSUE_TEMPLATE.md).
```
Comment (eddiejaoude) [inline CONTRIBUTING.md]: My suggestion would be to link to the `Issues` section, as it will be pre-populated with the issue template

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+Include screenshots and error log if available. 
+
+## In progress
+Chosen the issue you want to work on, let us know by tagging one of the project maintainers: *@thanosbnt*, 
+*@sassalley*, *@lorenaqendro*, *@arya-hemanshu* and change the issue tag from `help wanted` to `in progress`. Regular
+ update on its status is much appreciated, so that everyone knows its proceeding and eventually help.
```
Comment (eddiejaoude) [inline CONTRIBUTING.md]: - I don't think it is possible for `contributors` to change the issue labels.
- I think most will tag the first person or all, may be give each maintainer an area to look after?

---
e.g 

Once you have started to work on an issue, let us know by putting in a comment and tagging the relevant maintainer: 

| Maintainer | Section / Type of work |
| :--- | :--- |
| `@thanosbnt` | this section / type of work |
| `@sassalley` | this section / type of work |
| `@lorenaqendro` | this section / type of work |
| `@arya-hemanshu` | this section / type of work | 

Regular updates on the status will be useful to the team, and will allow us to not only provide support but also to understand the progress.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,103 @@
+# Contributing to Tombolo Digital Connector
+
+Thank you for choosing to contribute to this project. **We need your superpowers!**
```
Comment (eddiejaoude) [inline CONTRIBUTING.md]: > **We need your superpowers!** 

Love this 😄

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+## Choosing an issue and start contributing
+If you are new to github you might want to read more about [issues](https://help.github.com/articles/about-issues/)
+and [pull requests](https://help.github.com/articles/about-pull-requests/). 
+To contribute to the project follow these steps:
+- [Fork the repository](https://help.github.com/articles/fork-a-repo/).
+- [Choose the issue](https://github.com/FutureCitiesCatapult/TomboloDigitalConnector/milestone/22) you want to start
+working on, change its tag to `in progress` and tag a project maintainer to inform about it.
+- Use the Github issue to discuss on the requirements, ask for more information, update on the progress or upload 
+images related to the issue.
+- Create a pull request when you feel you have addressed the issue.
```
Comment (eddiejaoude) [inline CONTRIBUTING.md]: May be add a link for `pull request` to the docs https://help.github.com/articles/about-pull-requests/

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+*@sassalley*, *@lorenaqendro*, *@arya-hemanshu* and change the issue tag from `help wanted` to `in progress`. Regular
+ update on its status is much appreciated, so that everyone knows its proceeding and eventually help.
+
+## Choosing an issue and start contributing
+If you are new to github you might want to read more about [issues](https://help.github.com/articles/about-issues/)
+and [pull requests](https://help.github.com/articles/about-pull-requests/). 
+To contribute to the project follow these steps:
+- [Fork the repository](https://help.github.com/articles/fork-a-repo/).
+- [Choose the issue](https://github.com/FutureCitiesCatapult/TomboloDigitalConnector/milestone/22) you want to start
+working on, change its tag to `in progress` and tag a project maintainer to inform about it.
```
Comment (eddiejaoude) [inline CONTRIBUTING.md]: Similar to above comment, I don't think contributors can change the label on an issue

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+- All of your commits are atomic (one feature per commit).
+- Write a clear log message for the commit explaining the changes included in it.
+  ```bash
+  git commit -m "Summary description of the changes"
+  > Detailed list if many
+  > Detailed paragraph if the change is big and needs further explanation"
+  ```
+- Run all the tests locally in your IDE or via `gradle test`.
+- Check the Wercker CLI pipeline succeeded with your last changes.
+- You have merged the last version of master into your branch.
```
Comment (eddiejaoude) [inline CONTRIBUTING.md]: Use backtics around `master`.

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -0,0 +1,103 @@
...
+
+Now you are ready to open a PR by following this considerations:
+- Include a description of your changes and approach taken, if many create a list of changes.
+- Reference the issue number you are addressing.
+- Keep it brief and clear. We won't complain about short PRs. Promise! ;)
+- Assign it for a review to a project owner.
```
Comment (eddiejaoude) [inline CONTRIBUTING.md]: I don think they can do this as a public contributor. 

They probably won't know who to assign it to. Maintainers are probably best to assign it to the right person once they see a new PR appear.

File: CONTRIBUTING.md [hunk 8]
```diff
@@ -0,0 +1,103 @@
...
+
+## Testing
+Wercker CLI is configured to run tests on GitHub, so when you open a PR the test suite will run automatically as a build
+ which you can view in the merge box of the pull request. You can see the test output by clicking the `Details` link 
+ next to the build.
+
+To run the test suite locally, use `gradle test` from the project root. If you use IntelliJ JUnit test runner, find the 
+configuration [here](README.md#run-tests).
+
+## Conventions
```
Comment (eddiejaoude) [inline CONTRIBUTING.md]: 👍

**Comment by eddiejaoude** [comment]
Looks really good 👍 . A few small comments 🤓

**Comment by lorenaqendro** [comment]
@eddiejaoude thanks a lot for you comments. 🔝

File: CONTRIBUTING.md [hunk 9]
```diff
@@ -13,24 +13,23 @@ milestone contains issues we think are interesting for you to work on. You can f
 [`help wanted`](https://github.com/FutureCitiesCatapult/TomboloDigitalConnector/issues?q=is%3Aopen+is%3Aissue+label%3A%22help+wanted%22) 
 tag. Feel free to check the other milestones in case you find something that excites you more.
 If none of the issues reflects your problem/feature create a new one following the 
-[issue template](https://github.com/FutureCitiesCatapult/TomboloDigitalConnector/blob/master/.github/ISSUE_TEMPLATE.md).
+[issue template](https://github.com/FutureCitiesCatapult/TomboloDigitalConnector/issues/new).
```
Comment (eddiejaoude) [inline CONTRIBUTING.md]: Is the word `... template` need in the link name?

Comment (lorenaqendro) [reply CONTRIBUTING.md]: Not really! a leftover 💃

File: CONTRIBUTING.md [hunk 10]
```diff
@@ -0,0 +1,103 @@
+# Contributing to Tombolo Digital Connector
+
+Thank you for choosing to contribute to this project. **We need your superpowers!**  
+To learn more about the Tombolo vision checkout our [website](http://www.tombolo.org.uk/).
```
Comment (sassalley) [inline CONTRIBUTING.md]: Potentially change checkout to check out and leave an empty line above it for aesthetic reasons.

File: CONTRIBUTING.md [hunk 11]
```diff
@@ -0,0 +1,103 @@
...
+[Open Source Community - Digital Connector](https://github.com/FutureCitiesCatapult/TomboloDigitalConnector/milestone/22)
+milestone contains issues we think are interesting for you to work on. You can find them also through 
+[`help wanted`](https://github.com/FutureCitiesCatapult/TomboloDigitalConnector/issues?q=is%3Aopen+is%3Aissue+label%3A%22help+wanted%22) 
+tag. Feel free to check the other milestones in case you find something that excites you more.
+If none of the issues reflects your problem/feature create a new one following the 
+[issue template](https://github.com/FutureCitiesCatapult/TomboloDigitalConnector/blob/master/.github/ISSUE_TEMPLATE.md).
+Include screenshots and error log if available. 
+
+## In progress
+Chosen the issue you want to work on, let us know by tagging one of the project maintainers: *@thanosbnt*,
```
Comment (sassalley) [inline CONTRIBUTING.md]: It might be helpful to add "If you have.." chosen this issue.

File: CONTRIBUTING.md (hunk 2 above)
Comment (sassalley) [inline CONTRIBUTING.md]: "Regular update on its status is much appreciated, so that everyone knows its proceeding and eventually help." 

This is a great point - perhaps rephrase it to: "Regular updates on issue status are much appreciated. They enable others to see progress and may eventually facilitate help."

File: CONTRIBUTING.md [hunk 12]
```diff
@@ -0,0 +1,102 @@
...
+## Conventions
+We don't like to impose strict code conventions, but we must think about the people that will read and eventually 
+change it in the future. Try to follow as much as possible the 
+ [Java code conventions](http://www.oracle.com/technetwork/java/codeconvtoc-136057.html), you can't go wrong with 
+ them. As a general rule read the suggested conventions, follow what's already embedded in the code and most 
+ importantly, use common sense.  The following guidelines suggest some minimal effort to write a clear and readable 
+ code.  
+
+- Code
+  - Indent with 4 spaces, not tabs.
```
Comment (arya-hemanshu) [inline CONTRIBUTING.md]: I think as long as tabs are 4 spaces, it doesn't really matter what they use, as they can set number of spaces of a tab press

File: CONTRIBUTING.md [hunk 13]
```diff
@@ -0,0 +1,102 @@
...
+  - No trailing whitespace. Blank lines should not have empty spaces.
+  - End a file with a newline.
+  - Use `a = b` and not `a=b`, `x += 1` and not `x+=1`, ...
+  - Use `my_method(my_arg)` not `my_method( my_arg )`.
```
Comment (arya-hemanshu) [inline CONTRIBUTING.md]: `my_method(my_arg)` is more python style writing of methods, java usually follow lower camel case, i think it would be good if we could change it to `myMethod(myArg)`. I suggest this because when i read this, i got an impression that python style method naming is allowed here.

**Comment by lorenaqendro** [comment]
@sassalley @arya-hemanshu if you're happy with the changes, express your happiness by approving this PR :)
